Makefile

```diff
@@ -950,8 +954,13 @@ all_$(1)_clean: $$(addsuffix _clean, $$(filter bu_$(1), $$(BU_TARGETS)))
 all_$(1)_clean: $$(addsuffix _clean, $$(filter ef_$(1), $$(EF_TARGETS)))
 endef
 
-# Some boards don't use the bootloader
+ifeq ($(filter debug, $(FLIGHT_BUILD_CONF)), debug)
+# Don't build boards that overflow with debugging
+ALL_BOARDS     := $(filter-out $(NODEBUG_BOARDS), $(ALL_BOARDS))
```

Comment:
Should probably print a warning to stderr with the list of boards that won't be built.

Reply:
so it's not easy to constrain this to cases where it makes a difference... I did this:

```
mlyle@midnight:~/dRonin$ FLIGHT_BUILD_CONF=debug make -j4 fw_sparky2
Makefile:961: Removing targets with inoperable debug mode: sparky pikoblx omnibusf3 sprf3e
make[1]: Nothing to be done for 'first'.
make[1]: Nothing to be done for 'flightlib_armhardfp'.
make[1]: Nothing to be done for 'flightlib_armsoftfp'.
make[1]: Nothing to be done for 'uavobjects_armsoftfp'.
make[1]: Nothing to be done for 'uavobjects_armhardfp'.
 CC        [fw|spk2]  flight/Modules/ManualControl/failsafe_control.c
```